src/itensor.jl

```diff
@@ -1892,7 +1892,7 @@ diag(T::ITensor) = diag(tensor(T))
 
 mul!(C::ITensor, A::ITensor, B::ITensor, args...)::ITensor = contract!(C, A, B, args...)
 
-dot(A::ITensor, B::ITensor) = (dag(A) * B)[]
+dot(A::ITensor, B::ITensor) = NDTensors.cpu(dag(A) * B)[]
```

Comment:
I guess this is to avoid calling `getindex` on GPU? Maybe should be handled in `NDTensors`.